### PR TITLE
Deprecate MongoCollections#get

### PR DESCRIPTION
**Pull request by thll**
Deprecates `MongoCollections#get` in favor of `MongoCollections#collection`.

I didn't touch the search code because I didn't want to interfere with https://github.com/Graylog2/graylog2-server/pull/19456

/nocl
/prd https://github.com/Graylog2/graylog-plugin-enterprise/pull/7380

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog/scheduler/JobDefinitionDto.java [hunk 1]
```diff
@@ -64,6 +58,7 @@ public static Builder create() {
         @Id
         @ObjectId
         @JsonProperty(FIELD_ID)
+        @Nullable
         public abstract Builder id(String id);
```
Comment (bernd) [inline graylog2-server/src/main/java/org/graylog/scheduler/JobDefinitionDto.java:62]: The `id` parameter should be `Nullable`, not the return value of the `#id()` builder method.

```suggestion
        public abstract Builder id(@Nullable String id);
```

File: graylog2-server/src/main/java/org/graylog2/indexer/indexset/profile/IndexFieldTypeProfile.java [hunk 2]
```diff
@@ -17,20 +17,18 @@
...
-import org.mongojack.ObjectId;
 
 import javax.annotation.Nonnull;
-import javax.annotation.Nullable;
 
 /**
  * Represents profile as it is stored in Mongo.
  */
-public record IndexFieldTypeProfile(@JsonProperty(ID_FIELD_NAME) @Nullable @Id @ObjectId String id,
+public record IndexFieldTypeProfile(String id,
```
Comment (bernd) [inline graylog2-server/src/main/java/org/graylog2/indexer/indexset/profile/IndexFieldTypeProfile.java:28]: Does annotation inheritance work for records as well? It works for abstract auto-value classes. I'm not sure if it works for records as well. Please check. Thanks!

File: graylog2-server/src/test/java/org/graylog2/database/MongoCollectionsTest.java [hunk 3]
```diff
@@ -56,12 +54,12 @@ class MongoCollectionsTest {
     private MongoCollections collections;
     private EncryptedValueService encryptedValueService;
 
-    record Person(@JsonProperty("id") @Id @org.mongojack.ObjectId String id,
+    record Person(String id,
```
Comment (bernd) [inline graylog2-server/src/test/java/org/graylog2/database/MongoCollectionsTest.java:57]: Same as above. Please check if annotation inheritance works for records. Thanks!

File: graylog2-server/src/test/java/org/graylog2/database/MongoCollectionsTest.java [hunk 4]
```diff
@@ -71,11 +69,10 @@ record Secret(@JsonProperty("encrypted_value") EncryptedValue encryptedValue) {}
     record IgnoreTest(@JsonProperty("ignore_me_not") String ignoreMeNot,
                       @MongoIgnore @JsonProperty("ignore_me") String ignoreMe) {}
 
-    record IdGenerationTest(@Nullable @JsonProperty("id") @Id @org.mongojack.ObjectId String id) {}
+    record IdGenerationTest(String id) implements MongoEntity {}
 
     @JsonIgnoreProperties(ignoreUnknown = true)
-    record TimestampTest(@Nullable @JsonProperty("id") @Id @org.mongojack.ObjectId String id,
-                         @JsonProperty("timestamp") DateTime timestamp) {}
+    record TimestampTest(String id, @JsonProperty("timestamp") DateTime timestamp) implements MongoEntity {}
```
Comment (bernd) [inline graylog2-server/src/test/java/org/graylog2/database/MongoCollectionsTest.java:75]: Same as above. Please check if annotation inheritance works for records. Thanks!

**Review by bernd — CHANGES_REQUESTED** [review]

File: graylog2-server/src/main/java/org/graylog/scheduler/JobDefinitionDto.java (hunk 1 above)
Comment (thll) [reply graylog2-server/src/main/java/org/graylog/scheduler/JobDefinitionDto.java]: Thanks for spotting this. The annotation was a leftover from an experiment and not intended to be kept. I removed it again.

File: graylog2-server/src/test/java/org/graylog2/database/MongoCollectionsTest.java (hunk 3 above)
Comment (thll) [reply graylog2-server/src/test/java/org/graylog2/database/MongoCollectionsTest.java]: The record behaves correctly but it's not very intuitive why it does so. I brought the annotations back to make it more explicit.

**Pull request merged**